app/src/main/java/org/systers/mentorship/view/fragments/EditProfileFragment.kt

```diff
@@ -97,4 +111,19 @@ class EditProfileFragment: DialogFragment() {
         profileViewModel.successfulUpdate.removeObservers(activity!!)
         profileViewModel.successfulUpdate.value = null
     }
+
+    /**
+     * A function to perform input validation on the data
+     * entered by the user (Add more parameters as needed in the future)
+     * If there are no errors, the returned array will be empty.
+     * @param name The user's name
+     * @return Return an array of errors.
+     */
+    private fun validateProfileInput(name: String?): Array<EditProfileFragmentErrorStates> {
+        var errors = arrayOf<EditProfileFragmentErrorStates>()
+        if (name.isNullOrEmpty()) errors += EditProfileFragmentErrorStates.EmptyNameError
+        if (name?.length ?: 0 < 2) errors += EditProfileFragmentErrorStates.NameTooShortError
```

Comment:
should this be on integer resources instead of hardcoded in this code? cc @m-murad